compiler-plugin/src/main/java/io/ballerina/stdlib/rabbitmq/plugin/RabbitmqServiceValidator.java

```diff
@@ -99,45 +97,24 @@ private void validateAttachPoint(SyntaxNodeAnalysisContext context) {
         Optional<Symbol> symbol = semanticModel.symbol(serviceDeclarationNode);
         if (symbol.isPresent()) {
             ServiceDeclarationSymbol serviceDeclarationSymbol = (ServiceDeclarationSymbol) symbol.get();
-            Optional<ServiceAttachPoint> attachPoint = serviceDeclarationSymbol.attachPoint();
-            List<AnnotationSymbol> symbolList = serviceDeclarationSymbol.annotations();
-            if (attachPoint.isEmpty()) {
-                if (symbolList.isEmpty()) {
-                    context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.NO_ANNOTATION,
-                            DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
-                } else if (symbolList.size() > 1) {
-                    context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_ANNOTATION_NUMBER,
-                            DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
-                } else {
-                    validateAnnotation(symbolList.get(0), serviceDeclarationNode.location(), context);
-                }
-            } else {
-                if (attachPoint.get().kind() != ServiceAttachPointKind.STRING_LITERAL) {
-                    if (serviceDeclarationSymbol.annotations().isEmpty()) {
-                        context.reportDiagnostic(PluginUtils.getDiagnostic(
-                                CompilationErrors.INVALID_SERVICE_ATTACH_POINT,
-                                DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
-                    } else {
-                        validateAnnotation(symbolList.get(0), serviceDeclarationNode.location(), context);
-                    }
-                }
+            Optional<ServiceAttachPoint> serviceNameAttachPoint = serviceDeclarationSymbol.attachPoint();
+            List<AnnotationSymbol> annotations = serviceDeclarationSymbol.annotations();
+
+            // RabbitMQ gets the subject name from either the service name or the
+            // service config, so either one of them should be present.
+            if (serviceNameAttachPoint.isEmpty() && annotations.isEmpty()) {
```

Comment:
What if the `serviceNameAttachPoint` is empty and there is a `@display` annotation, without the service config? It should be an error, but this does not catch it, right?

Reply:
hmm, yeah have to add a validation to check if the annotation is service config

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/874/commits/99cee71894b3171feb63b5e5b96c6e75342d695c